apps/google-analytics-4/lambda/src/lib/verify-signed-request-middleware.spec.ts

```diff
@@ -0,0 +1,43 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import sinon from 'sinon';
+import * as NodeAppsToolkit from '@contentful/node-apps-toolkit';
+import app from '../app';
+
+chai.use(chaiHttp);
+
+describe('verifySignedRequestMiddleware', () => {
+  const verifyRequestStub = sinon.stub();
+
+  beforeEach(() => {
+    // TODO: set headers and fully test signature verification later
+    sinon.stub(NodeAppsToolkit, 'verifyRequest').get(() => verifyRequestStub);
+  });
+
+  describe('given an unsigned request', () => {
+    before(() => {
+      verifyRequestStub.returns(false);
+    });
+
+    it('returns 403 Unauthorized', async () => {
+      const response = await chai.request(app).get('/api/credentials');
```

Comment:
We will probably want to refactor this test or figure out another way to integration test it. E.g. here I'd like to remove the /api/credentials endpoint, but the presence of this line will block that.

(I ran into the same testing question when I considred writing integration tests for our mapper)

Reply:
We could do what I did in `apiErrorMapper` and test the middleware directly btw.

Reply:
Yeah i had it like that at first, doing a mock app instance, but i wasn't a fan of that because we have 3 middlewares all working together on the full response expectation for this integration test. Seems valuable to just use a real route and change the path if that endpoint goes away.

Reply:
discussed offline and will keep like this for now, but consider adding more isolated unit tests for compartmentalization later on